web/src/stats/stats.ts

```diff
@@ -517,15 +609,19 @@ function compute_summary_chart_data(time_series_data, num_steps, labels_) {
     };
 }
 
-function populate_messages_sent_by_client(data) {
+function populate_messages_sent_by_client(data: SentData): void {
     // Content rendered by this method is titled as "Messages sent by client" on the webpage
     
-    const layout = {
+    type PlotDataByMessageClient = PlotTrace & {
+        trace_annotations: Partial<Plotly.PlotData>;
+    };
+
+    const layout: Partial<Plotly.Layout> = {
         width: 750,
-        height: null, // set in draw_plot()
+        height: undefined, // set in draw_plot()
```

Comment:
`null` is incompabitle with the types provided by `Plotly` library. Only `number | undefined` is accepted. So thats why here I changed from `null` to undefined. Same reason for `xaxis: {range: undefined}` three lines below